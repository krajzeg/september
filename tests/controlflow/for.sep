print("This should print 1, 2, 3.")
for (number) in ([1,2,3]) {
	print(number)
}

print("This should never execute the body.")
for (number) in ([]) {
	print("But it did.")
}

print("This should print even numbers up to 6.")
for (number) in ([1,2,3,4,5,6,7,8]) {
	if (number > 6) { break }
	if (number % 2 != 0) { continue }
	print(number)
}

print("Exceptions from within 'for' should be propagated.")
try {
	for (number) in ([1]) {
		missing_prop = 42
	}
	print("But they aren't.")
} catch(EMissingProperty) {
	print("And they are.")
}
