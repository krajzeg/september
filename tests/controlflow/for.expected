This should print 1, 2, 3.
1
2
3
This should never execute the body.
This should print even numbers up to 6.
2
4
6
